homeassistant/components/sensor/__init__.py

```diff
@@ -399,7 +399,12 @@ def state(self) -> Any:
         native_unit_of_measurement = self.native_unit_of_measurement
         unit_of_measurement = self.unit_of_measurement
         value = self.native_value
-        device_class = self.device_class
+        device_class: SensorDeviceClass | None = None
+        with suppress(ValueError):
```

Comment:
Don't we want the logging so that users are informed about the invalid `device_class`?